pipeline/src/node_properties/function_properties.rs

```diff
@@ -1,8 +1,11 @@
 use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use std::collections::HashMap;
 
 #[derive(Clone, Debug, Deserialize, PartialEq, Serialize)]
 pub struct FunctionProperties {
     pub code: String,
+    pub props: HashMap<String, Value>,
```

Comment:
`HashMap` adds a weird kind of non-determinism to JSON serialization, I'd prefer if we used `serde_json::Map<String, Value>` or `BTreeMap<String, Value>`.

Reply:
I trust you blindly on all things serde but just for my curiosity, what is the issue?

Reply:
Since HashMap iteration order is non-deterministic (depends on a random seed obtained at HashMap creation time), serialization order of HashMap entries is also non-deterministic.

Reply:
ah the order of entries. Does that matter to us here?

Reply:
It usually doesn't matter, but I dislike equal objects serializing differently on principle.